numba/tests/test_withlifting.py

```diff
@@ -945,6 +945,56 @@ def case_objmode_cache(x):
             output = x / 10
         return output
 
+    def test_objmode_reflected_list(self):
+        ret_type = typeof([1, 2, 3, 4, 5])
+        @njit
+        def test2():
+            with objmode(out=ret_type):
+                out = [1, 2, 3, 4, 5]
+            return out
+
+        with self.assertRaises(errors.CompilerError) as raises:
+            test2()
+        self.assertIn(
+            ("Objmode context failed. "
+             "Argument 'out' is using unsupported type. "
+             "Reflected types is not supported."),
+            str(raises.exception),
+        )
+
+    def test_objmode_reflected_set(self):
+        ret_type = typeof({1, 2, 3, 4, 5})
+        @njit
+        def test2():
+            with objmode(result=ret_type):
+                result = {1, 2, 3, 4, 5}
+            return result
+
+        with self.assertRaises(errors.CompilerError) as raises:
+            test2()
+        self.assertIn(
+            ("Objmode context failed. "
+             "Argument 'result' is using unsupported type. "
+             "Reflected types is not supported."),
+            str(raises.exception),
+        )
+
+    def test_objmode_typed_dict(self):
+        ret_type = types.DictType(types.unicode_type, types.int64)
+        @njit
+        def test4():
+            with objmode(res=ret_type):
+                res = {'A': 1, 'B': 2}
+            return res
+
+        with self.assertRaises(TypeError) as raises:
+            test4()
+        self.assertIn(
+            ("can't unbox <class 'numba.typed.typeddict.Dict'> "
+             "from <class 'dict'>"),
```

Comment:
Am I right in thinking that the issue here is that a CPython dictionary cannot be unboxed to a `typed.Dict`, and more generally that the returned object will not unbox as the declared type? If so, would something like `"Cannot unbox a <class 'dict'> as a <class 'numba.typed.typeddict.Dict'>"` be more clear?

Reply:
Yes, i'll change that.